Bottom: 7e8d54516318cd806149d7b70d58b2e8b9ae4a1f
Top:    96cc1f97e4aee2e4c2cc9e1822033fd81b8ebca3
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-03 12:56:56 +0800

Refresh of vhost-packed-ring-support

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 1c295dd..f85a3f3 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2635,7 +2635,8 @@ static int vhost_add_used_packed(struct vhost_virtqueue *vq,
 
 	if (idx == vq->last_used_idx) {
 		/* Make sure descriptor id and len is written before
-		   flags for the first used buffer. */
+		 * flags for the first used buffer.
+		 */
 		smp_wmb();
 	}
 
@@ -2669,6 +2670,9 @@ static int vhost_add_used_n_packed(struct vhost_virtqueue *vq,
 	u16 wrap_counter = vq->last_used_wrap_counter;
 	int i, ret;
 
+	/* Update used elems other than first to save unnecessary
+	 * memory barriers.
+	 */
 	for (i = 1; i < count; i++) {
 		if (last_used_idx >= vq->num) {
 			last_used_idx -= vq->num;
